clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/ApplicationCachingAgent.java

```diff
@@ -0,0 +1,97 @@
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.ecs.provider.agent;
+
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE;
+import static com.netflix.spinnaker.clouddriver.ecs.cache.Keys.Namespace.APPLICATIONS;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.ecs.AmazonECS;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.cats.agent.AgentDataType;
+import com.netflix.spinnaker.cats.cache.CacheData;
+import com.netflix.spinnaker.cats.provider.ProviderCache;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.cache.model.Application;
+import java.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ApplicationCachingAgent extends AbstractEcsOnDemandAgent<Application> {
+  private static final Collection<AgentDataType> types =
+      Collections.unmodifiableCollection(
+          Arrays.asList(AUTHORITATIVE.forType(APPLICATIONS.toString())));
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  private ObjectMapper objectMapper;
+
+  public ApplicationCachingAgent(
+      NetflixAmazonCredentials account,
+      String region,
+      AmazonClientProvider amazonClientProvider,
+      AWSCredentialsProvider awsCredentialsProvider,
+      Registry registry,
+      ObjectMapper objectMapper) {
+    super(account, region, amazonClientProvider, awsCredentialsProvider, registry);
+    this.objectMapper = objectMapper;
+  }
+
+  public static Map<String, Object> convertApplicationToAttributes(Application application) {
+    Map<String, Object> attributes = new HashMap<>();
+    attributes.put("name", application.getName());
+    return attributes;
+  }
+
+  @Override
+  public Collection<AgentDataType> getProvidedDataTypes() {
+    return types;
+  }
+
+  @Override
+  public String getAgentType() {
+    return accountName + "/" + region + "/" + getClass().getSimpleName();
+  }
+
+  @Override
+  protected List<Application> getItems(AmazonECS ecs, ProviderCache providerCache) {
+    List<Application> applications = new ArrayList<>();
+    return applications;
+  }
+
+  @Override
+  protected Map<String, Collection<CacheData>> generateFreshData(
+      Collection<Application> applications) {
+    Collection<CacheData> applicationData = new LinkedList<>();
+
+    //    for (Application application : applications) {
+    //      Map<String, Object> attributes = convertApplicationToAttributes(application);
+    //      String applicationKey = Keys.getApplicationKey(application.getName());
+    //
+    //      applicationData.add(
+    //          new DefaultCacheData(applicationKey, attributes, application.getRelationships()));
+    //    }
```

Comment:
We can remove this commented part if we don't need this in the future.

Reply:
Good catch, I meant to uncomment that